highlevel/adapter/socket/isotp.py

```diff
@@ -0,0 +1,87 @@
+"""
+CAN ISO-TP socket adapter module.
+"""
+import asyncio
+import socket
+from asyncio.protocols import DatagramProtocol
+from asyncio.transports import DatagramTransport, BaseTransport
+from dataclasses import dataclass
+from typing import List, Optional, Tuple
+
+from highlevel.adapter.socket import SocketAdapter, CallbackFunc
+from highlevel.logger import LOGGER
+
+
+@dataclass
+class ISOTPAddress:
+    """
+    Container for ISOTP addressing data.
+    """
+    device: str  # CAN device to use. Can be virtual.
+    id_reception: int  # CAN id to listen on.
+    id_transmission: int  # CAN id to transmit to.
+
+
+class ISOTPSocketAdapter(SocketAdapter, DatagramProtocol):
+    """
+    Implements an ISO-TP compatible adapter using the can-isotp driver.
+    Sets up a datagram endpoint with asyncio.
+    """
+    def __init__(self, address: ISOTPAddress, adapter_name: str):
+        self.socket = socket.socket(socket.AF_CAN, socket.SOCK_DGRAM,
+                                    socket.CAN_ISOTP)
+        self.callbacks: List[CallbackFunc] = []
+        self.address = (address.device, address.id_reception,
+                        address.id_transmission)
+        self.adapter_name = adapter_name
+        self.transport: Optional[DatagramTransport] = None
+        self.protocol: Optional[DatagramProtocol] = None
+
+    async def run(self) -> None:
+        """
+        Binds the internal UDP socket and Starts the receiver task.
+        Does not block after it is done.
+        """
+        try:
+            self.socket.bind(self.address)
+            transport, protocol = await asyncio.get_event_loop(
+            ).create_datagram_endpoint(sock=self.socket,
+                                       protocol_factory=lambda: self)
+            # DatagramProtocol methods are typed with BaseTransport transports, but are actually
+            # DatagramTransports, that have methods we need here. So we ignore the mypy errors.
+            self.transport = transport  # type: ignore
+            self.protocol = protocol  # type: ignore
+            while True:
+                await asyncio.sleep(1)
+
+        except ConnectionRefusedError:
+            raise RuntimeError("Error on datagram endpoint creation:" +
```

Comment:
I would just raise? or not catch? 

and add logging, just so we don't "eat" the exception
![image](https://user-images.githubusercontent.com/36550428/83577129-bcf04700-a533-11ea-8d37-99a97a25e49a.png)